applications/StructuralMechanicsApplication/custom_response_functions/adjoint_processes/replace_multiple_elements_and_conditions_process.cpp

```diff
@@ -0,0 +1,115 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Armin Geiser
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "replace_multiple_elements_and_conditions_process.h"
+#include "utilities/compare_elements_and_conditions_utility.h"
+
+namespace Kratos
+{
+
+namespace {
+
+template <class TEntity, class TEntityContainer>
+void ReplaceEntities(TEntityContainer& rEntityContainer,
+                     Parameters EntitySettings,
+                     bool ThrowError)
+{
+    // create map with reference entities
+    std::map<std::string, const TEntity*> entities_table;
+    for(Parameters::iterator it=EntitySettings.begin(); it!=EntitySettings.end(); ++it){
+        entities_table[it.name()] = &KratosComponents<TEntity>::Get(EntitySettings[it.name()].GetString());
+    }
+
+    #pragma omp parallel for
+    for(int i=0; i< static_cast<int>(rEntityContainer.size()); i++) {
+        auto it = rEntityContainer.begin() + i;
+
+        std::string current_name;
+        CompareElementsAndConditionsUtility::GetRegisteredName(*it, current_name);
+
+        auto it_reference_entity = entities_table.find(current_name);
+
+        if (it_reference_entity == entities_table.end()) {
+            // This error is thrown in a parallel region and can not get catched or even printed properly!
+            KRATOS_ERROR_IF(ThrowError) << current_name
+                << " was not defined in the replacement table!" << std::endl;
+            continue; // skip if no error should be thrown
+        }
+
+        auto p_entitiy = it_reference_entity->second->Create(it->Id(),
+            it->pGetGeometry(),
+            it->pGetProperties()
+        );
+
+        // Deep copy data and flags
+        p_entitiy->Data() = it->Data();
+        p_entitiy->Set(Flags(*it));
+
+        (*it.base()) = p_entitiy;
+    }
+}
+
+} // namespace
+
+
+void ReplaceMultipleElementsAndConditionsProcess::Execute()
+{
+    ModelPart& r_root_model_part = mrModelPart.GetRootModelPart();
+
+    bool throw_error = mSettings["throw_error"].GetBool();
+
+    // replace elements
+    ReplaceEntities<Element, ModelPart::ElementsContainerType>(mrModelPart.Elements(), mSettings["element_name_table"], throw_error);
```

Comment:
```suggestion
    ReplaceEntities(mrModelPart.Elements(), mSettings["element_name_table"], throw_error);
```

This should work

`TEntity` should be retrievable from TEntityContainer

Reply:
worked, thanks